scripts/gdb-pretty-printers.py

```diff
@@ -0,0 +1,152 @@
+            self.buffer_start = 0
+            # At most 1 MB or size, at least 1
+            self.buffer_size = min(size, max(1, 2 ** 20 // self.sizeof))
+            self.buffer = gdb.parse_and_eval(
+                '(void*)malloc(%s)' % (self.buffer_size * self.sizeof))
+            self.buffer.fetch_lazy()
+            self.buffer_count = self.buffer_size
+            self.update_buffer()
+
+        def update_buffer(self):
+            if self.buffer_count >= self.buffer_size:
+                self.buffer_item = gdb.parse_and_eval(
+                    hex(self.buffer)).cast(self.item.type)
+                self.buffer_count = 0
+                self.buffer_start = self.count
+                device_addr = hex(self.item.dereference().address)
+                buffer_addr = hex(self.buffer)
+                size = min(self.buffer_size, self.size -
+                            self.buffer_start) * self.sizeof
+                status = gdb.parse_and_eval(
+                    '(cudaError)cudaMemcpy(%s, %s, %d, cudaMemcpyDeviceToHost)' % (buffer_addr, device_addr, size))
+                if status != 0:
+                    raise gdb.MemoryError(
+                        'memcpy from device failed: %s' % status)
+
+        def __del__(self):
+            gdb.parse_and_eval('(void)free(%s)' %
+                                hex(self.buffer)).fetch_lazy()
+
+        def __iter__(self):
+            return self
+
+        def __next__(self):
+            if self.count >= self.size:
+                raise StopIteration
+            self.update_buffer()
+            elt = self.buffer_item.dereference()
+            self.buffer_item = self.buffer_item + 1
+            self.buffer_count = self.buffer_count + 1
+            count = self.count
+            self.item = self.item + 1
+            self.count = self.count + 1
+            return ('[%d]' % count, elt)
+
+
+    class RmmDeviceUVectorPrinter(gdb.printing.PrettyPrinter):
+        """Print a rmm::device_uvector"""
+
+        def __init__(self, val):
+            self.val = val
+            el_type = val.type.template_argument(0)
+            self.pointer = val['_storage']['_data'].cast(el_type.pointer())
+            self.size = int(val['_storage']['_size']) // el_type.sizeof
+            self.capacity = int(val['_storage']['_capacity']) // el_type.sizeof
+
+        def children(self):
+            return DeviceIterator(self.pointer, self.size)        
+
+        def to_string(self):
+            typename = str(self.val.type)
+            return ('%s of length %d, capacity %d' % (typename, self.size, self.capacity))
+
+        def display_hint(self):
+            return 'array'
+
+
+    # Workaround to avoid using the pretty printer on things like std::vector<int>::iterator
+    def is_template_type_not_alias(typename):
+        loc = typename.find('<')
+        if loc is None:
+            return False
+        depth = 0
+        for char in typename[typename.find('<'):-1]:
```

Comment:
Reuse the value found before.
```suggestion
        for char in typename[loc:-1]:
```